### ISSUE TITLE
Ruff fails on the issue introduced during fix

### ISSUE THREAD

**Issue by Cjkjvfnby** [astral-sh/ruff#5150]
Applying FLY002 to the code creates a code that does not pass `E501 Line too long`

This check is mentioned in the https://github.com/astral-sh/ruff/issues/2102

Code sample:
```python
expected = "\n".join(
    [
        "=== long text ======================================================================",
        "=== long text ======================================================================",
    ],
)
```

```shell
ruff scratch.py
scratch.py:1:12: FLY002 [*] Consider `"=== long text ======================================================================\n=== long text ======================================================================"` instead of string join    
Found 1 errors.
[*] 1 potentially fixable with the --fix option.
```

```shell
ruff scratch.py --fix
scratch.py:1:99: E501 Line too long (183 > 98 characters)
Found 1 errors (1 fixed, 0 remaining).
```

**Comment by MichaReiser** [comment]
I think in this case it's best to use a noqa comment. We used to consider the line length for some rules but this can lead to undesired false-negatives because it's far from a good metric for complexity. 

Another alternative here, but it requires human intervention, is to use a multiline string instead.
